wsl-pro-service/cmd/wsl-pro-service/service/config.go

```diff
@@ -0,0 +1,96 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	log "github.com/canonical/ubuntu-pro-for-wsl/common/grpc/logstreamer"
+	"github.com/canonical/ubuntu-pro-for-wsl/common/i18n"
+	"github.com/canonical/ubuntu-pro-for-wsl/wsl-pro-service/internal/consts"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/ubuntu/decorate"
+)
+
+func initViperConfig(name string, cmd *cobra.Command, vip *viper.Viper) (err error) {
+	defer decorate.OnError(&err, "can't load configuration")
+
+	// Use command-line flag for verbosity until configuration is parsed
+	v, err := cmd.Flags().GetCount("verbosity")
+	if err != nil {
+		return fmt.Errorf("internal error: no persistent verbosity flags installed on cmd: %w", err)
+	}
+	setVerboseMode(v)
+
+	// Find a valid configuration file
+	if v, err := cmd.Flags().GetString("config"); err == nil && v != "" {
+		vip.SetConfigFile(v)
+	} else {
+		vip.SetConfigName(name)
+		vip.AddConfigPath("./")
+		vip.AddConfigPath("$HOME/")
+		vip.AddConfigPath("/etc/")
+		if binPath, err := os.Executable(); err != nil {
+			log.Warningf(context.Background(), "Failed to get the current executable path, not adding it as a config dir: %v", err)
+		} else {
+			vip.AddConfigPath(filepath.Dir(binPath))
+		}
+	}
+
+	// Load the config
+	if err := vip.ReadInConfig(); err != nil {
+		var e viper.ConfigFileNotFoundError
+		if errors.As(err, &e) {
+			log.Infof(context.Background(), "No configuration file: %v", e)
+		} else {
+			return fmt.Errorf("invalid configuration file: %v", err)
+		}
+	} else {
+		log.Infof(context.Background(), "Using configuration file: %v", vip.ConfigFileUsed())
```

Comment:
This one shouldn't be hard to exercise: add a test similar to `TestNoUsageError` that writes a configuration file in one of the supported locations setting the verbosity level to the maximum and assert the configuration file path is present in the captured stdout. I don't think we need a whole table testing all possible supported locations, as that would be testing Viper, instead of testing our connecting logic.

Reply:
The config won't be used at this point, so this log would rely on the presence of `-v` instead until it is. If you think it's still worthwhile, I can still test for the log output while passing in `-v`

Reply:
You can add `-v` and have the config file set verbosity to 3, to prove it works and assert that the captured stdout refers to it.
Additionally, you can set an invalid config and assert for errors. For inspiration:

```go
func TestConfigVerbosity(t *testing.T) {
	getStdout := captureStdout(t)

	sys, _ := testutils.MockSystem(t)
	a := service.New(service.WithSystem(sys))
	a.SetArgs("version", "-v")

	filename := "wsl-pro-service.yaml"
	configNextToBinaryPath := filepath.Join(filepath.Dir(os.Args[0]), filename)
	require.NoError(t, os.WriteFile(configNextToBinaryPath, []byte("invalid-config"), 0644), "Couldn't write config file")
	// require.NoError(t, os.WriteFile(configNextToBinaryPath, []byte("verbosity: 3"), 0644), "Couldn't write config file")


	err := a.Run()
	out := getStdout()
	require.Error(t, err, "Run should not return an error, stdout: %v", out)
	require.Contains(t, out, filename, "Config file should be read. Output was\n", out)
}
```

and 

https://github.com/ubuntu/authd/blob/main/cmd/authd/daemon/daemon_test.go#L278-L334